zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/NotebookRepo.java

```diff
@@ -31,6 +32,9 @@
  * Notebook repository (persistence layer) abstraction
  */
 public interface NotebookRepo {
+
+  void init(ZeppelinConfiguration zConf) throws IOException;
```

Comment:
Java SPI require a non-argument constructor, so introduce the init method for initialization instead of putting it in constructor.